client/status/common.go

```diff
@@ -53,3 +55,71 @@ func (x ServerInternal) Message() string {
 func WriteInternalServerErr(x *ServerInternal, err error) {
 	x.SetMessage(err.Error())
 }
+
+// WrongMagicNumber describes failure status related to incorrect network magic.
+// Instances provide Status and StatusV2 interfaces.
+type WrongMagicNumber struct {
+	v2 status.Status
+}
+
+func (x WrongMagicNumber) Error() string {
+	return errMessageStatusV2(
+		globalizeCodeV2(status.WrongMagicNumber, status.GlobalizeCommonFail),
+		x.v2.Message(),
+	)
+}
+
+// implements method of the FromStatusV2 local interface.
+func (x *WrongMagicNumber) fromStatusV2(st *status.Status) {
+	x.v2 = *st
+}
+
+// ToStatusV2 implements StatusV2 interface method.
+// If the value was returned by FromStatusV2, returns the source message.
+// Otherwise, returns message with
+//  * code: WRONG_MAGIC_NUMBER;
+//  * string message: empty;
+//  * details: empty.
+func (x WrongMagicNumber) ToStatusV2() *status.Status {
+	x.v2.SetCode(globalizeCodeV2(status.WrongMagicNumber, status.GlobalizeCommonFail))
+	return &x.v2
+}
+
+// WriteCorrectMagic writes correct network magic.
+func (x *WrongMagicNumber) WriteCorrectMagic(magic uint64) {
+	// serialize the number
+	buf := make([]byte, 8)
+
+	binary.BigEndian.PutUint64(buf, magic)
```

Comment:
Maybe `LittleEndian`? It is more common in NEO world and matches endianness of most popular arches.

Reply:
It also requires change in [here](https://github.com/nspcc-dev/neofs-api/blob/e09a560d5781718a4d1eb27c89435958610bc600/status/types.proto#L86).

Reply:
`BigEndian` is specified in protocol, this code just followsit. You can create an issue in API repo